src/passes/I64ToI32Lowering.cpp

```diff
@@ -516,13 +516,19 @@ struct I64ToI32Lowering : public WalkerPass<PostWalker<I64ToI32Lowering>> {
 
   void lowerEqZInt64(Unary* curr) {
     TempVar highBits = fetchOutParam(curr->value);
-    replaceCurrent(
-      builder->makeBinary(
-        AndInt32,
-        builder->makeUnary(EqZInt32, builder->makeGetLocal(highBits, i32)),
-        builder->makeUnary(EqZInt32, curr->value)
+
+    Block* result = builder->blockify(
```

Comment:
no need for the block here, can just do `auto* result = ` and then the `makeUnary`

Reply:
this kind of gets to my point, that when the low bits are just used once, the situation is much simpler - no need for a temp, no need for a block, which makes it shorter to write and at least in my opinion easier to understand.

but reading the other cases, i do see your point that in many or most you do need to read the low bits more than once, in which case for consistency maybe it's better for this to be that way too. i don't feel strongly either way, up to you.